front_end/src/app/child-checkin/child-checkin.component.ts

```diff
@@ -10,17 +14,76 @@ import { MachineConfiguration } from '../setup/machine-configuration';
 })
 @Injectable()
 export class ChildCheckinComponent implements OnInit {
+  @ViewChild('serviceSelectModal') public serviceSelectModal: ModalDirective;
+  @ViewChild('childDetailModal') public childDetailModal: ModalDirective;
   private kioskDetails: MachineConfiguration;
 
-  constructor(private setupService: SetupService) {
+  clock = Observable.interval(10000).map(() => new Date());
+  thisSiteName: string;
+  selectedEvent: Event;
+  todaysEvents: Event[];
+  ready: boolean;
+
+  constructor(private setupService: SetupService, private adminService: AdminService) {
```

Comment:
Lets pull admin service stuff out of here and put the stuff used app wide in an app level service or shared service.  Then we can inject that.  Don't like injecting a service from another module into this module.